File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    let x = 0, y = 0;
    for(let key of keyinput){
        if (key === "left" && x > -Math.trunc(board[0]/2)){
            x -= 1
        } else if (key === "right" && x < Math.trunc(board[0]/2)){
            x += 1
        } else if (key === "up" && y < Math.trunc(board[1]/2)){
            y += 1
        } else if(key === "down" && y > -Math.trunc(board[1]/2)){
            y -= 1
        }
    }
    
    return [x,y];
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    x,y = 0,0
    moves = {"up": [0,1], "down": [0,-1], "left": [-1,0], "right": [1,0]}
    w,h = board[0]//2, board[1]//2
    
    def isValid(x,y, dx,dy):
        return -w <= x+dx <= w and -h <= y+dy <= h
    
    for k in keyinput:
        dx,dy = moves[k]
        if isValid(x,y, dx,dy):
            x += dx
            y += dy
            
    return [x,y]